ndcube/mixins/ndslicing.py

```diff
@@ -31,31 +33,62 @@ def __getitem__(self, item):
         return super().__getitem__(item)
 
     def _slice(self, item):
-        """
-        Construct a set of keyword arguments to initialise a new (sliced)
+        """Construct a set of keyword arguments to initialise a new (sliced)
         instance of the class. This method is called in
         `astropy.nddata.mixins.NDSlicingMixin.__getitem__`.
 
-        This method extends the `~astropy.nddata.mixins.NDSlicingMixin`
-        method to add support for ``missing_axes`` and ``extra_coords``
-        and overwrites the astropy handling of wcs slicing.
+        This method extends the `~astropy.nddata.mixins.NDSlicingMixin` method
+        to add support for  ``extra_coords`` and overwrites the astropy
+        handling of wcs slicing.
+
+        Parameters
+        ----------
+        item : slice
+            The slice passed to ``__getitem__``. Note that the item parameter corresponds
+            to numpy ordering, keeping with the convention for NDCube.
+
+        Returns
+        -------
+        dict :
+            Containing all the attributes after slicing - ready to
+            use them to create ``self.__class__.__init__(**kwargs)`` in
+            ``__getitem__``.
         """
+
+        item = tuple(sanitize_slices(item, len(self.dimensions)))
         kwargs = super()._slice(item)
 
-        wcs, missing_axes = self._slice_wcs_missing_axes(item)
-        kwargs['wcs'] = wcs
-        kwargs['missing_axes'] = missing_axes
-        kwargs['extra_coords'] = self._slice_extra_coords(item, missing_axes)
+        # Store the original dimension of NDCube object before slicing
+        prev_dim = len(self.dimensions)
+
+        # Sanitize the input arguments
+        wcs = self._slice_wcs(item)
 
+        # Set the kwargs values
+        kwargs['wcs'] = wcs
+        kwargs['extra_coords'] = self._slice_extra_coords(item, wcs._pixel_keep, prev_dim)
         return kwargs
 
-    def _slice_wcs_missing_axes(self, item):
-        # here missing axis is reversed as the item comes already in the reverse order
-        # of the input
-        return utils.wcs._wcs_slicer(
-            self.wcs, copy.deepcopy(self.missing_axes[::-1]), item)
+    def _slice_wcs(self, item):
```

Comment:
Can this be removed and we use the inherited ```NDData``` method?

What's the consequence of this returning a ```LowLevelSlicedWCS``` instead of a ```HighLevelSlicedWCS```?